lib/preface_specs/lattice.mli

```diff
@@ -0,0 +1,41 @@
+(** A [Lattice] capture the notion of a join semilattice and a meet semilattice *)
+
+(** {1 Minimal definition}*)
+
+module type WITH_JOIN_SEMILATTICE_AND_MEET_SEMILATTICE = sig
```

Comment:
To add context to @xhtmlboi's suggestion, here you describe a _requirement_ that requires the presence of two functions. When you want to build an abstraction based on one (or more) existing abstraction(s), you don't need to specify a _requirement_ because you can easily express the functor that will describe it using the existing interfaces.

Reply:
Thanks for the explanation !